### PR TITLE
[full-ci][tests-only] add a full CI and early fail support for the e2e test pipelines

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- Add config dict for e2e test pipes
- Early fail on e2e test pipes
- If full-ci then, no early fail on e2e test pipes

## Related Issue
- Fixes https://github.com/owncloud/web/issues/7547

## How Has This Been Tested?
- :robot: 
- with no `full-ci` and an intentional error on the e2e pipes: https://drone.owncloud.com/owncloud/web/27928
- with `full-ci` in the PR title and an intentional failure in the e2e suite: https://drone.owncloud.com/owncloud/web/27927
- with `skip` set as True: https://drone.owncloud.com/owncloud/web/27925

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
with `skip` set as True: https://drone.owncloud.com/owncloud/web/27925

**Comment by kiranparajuli589** [comment]
with `full-ci` in the PR title and an intentional failure in the e2e suite: https://drone.owncloud.com/owncloud/web/27927

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/27927/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27927/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-8-31-09-54-29.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27927/tracing/re-sharing-alice-2022-8-31-09-52-46.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27927/tracing/re-sharing-brian-2022-8-31-09-52-52.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27927/tracing/re-sharing-carol-2022-8-31-09-52-58.zip 


</p></details>

**Comment by kiranparajuli589** [comment]
with no `full-ci` and an intentional error on the e2e pipes: https://drone.owncloud.com/owncloud/web/27928

File: .drone.star [hunk 1]
```diff
@@ -823,7 +827,7 @@ def stagePipelines(ctx):
     unit_test_pipelines = unitTests(ctx)
     e2e_pipelines = e2eTests(ctx)
     acceptance_pipelines = acceptance(ctx)
-    return unit_test_pipelines + pipelinesDependsOn(e2e_pipelines, unit_test_pipelines) + pipelinesDependsOn(acceptance_pipelines, e2e_pipelines)
+    return unit_test_pipelines + pipelinesDependsOn(e2e_pipelines, unit_test_pipelines) + pipelinesDependsOn(acceptance_pipelines, unit_test_pipelines)
```
Comment (grgprarup) [inline .drone.star:830]:
```suggestion
    return unit_test_pipelines + pipelinesDependsOn([e2e_pipelines, acceptance_pipelines], unit_test_pipelines)
```

File: .drone.star [hunk 2]
```diff
@@ -1187,25 +1207,25 @@ def e2eTests(ctx):
...
-            "REPORT_TRACING": reportTracing,
+            "REPORT_TRACING": params["reportTracing"],
         },
         "commands": [
             "sleep 10 && yarn test:e2e:cucumber tests/e2e/cucumber/**/*[!.ocis].feature",
         ],
     }]
 
-    services = databaseService(db) + owncloudService() + webService()
+    services = databaseService(default["db"]) + owncloudService() + webService()
```
Comment (ScharfViktor) [inline .drone.star:1217]: should be params["db"] ?

Comment (kiranparajuli589) [reply .drone.star]: true, fixed :rocket:

**Comment by kiranparajuli589** [comment]
@saw-jan will be looking on this from now.

**Review by ScharfViktor — APPROVED** [review]
LGTM

File: tests/e2e/cucumber/features/integrations/link.feature [hunk 3]
```diff
@@ -4,7 +4,7 @@ Feature: link
     Given "Admin" creates following users
       | id    |
       | Alice |
-    When "Alice" logs in
+    When "Alic" logs in
```
Comment (individual-it) [inline tests/e2e/cucumber/features/integrations/link.feature:7]: :question:

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/27957/12/1
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

File: tests/e2e/cucumber/features/integrations/link.feature (hunk 3 above)
Comment (saw-jan) [reply tests/e2e/cucumber/features/integrations/link.feature:7]: I am checking the workflow :sweat_smile:

**Comment by saw-jan** [comment]
> Results for **e2e-tests oCIS** https://drone.owncloud.com/owncloud/web/27957/12/1 boom The e2e-ocis tests pipeline failed. The build has been cancelled.

Early fail comment working : https://drone.owncloud.com/owncloud/web/27957

**Review by individual-it — APPROVED** [review]

**Comment by saw-jan** [comment]
> Results for **e2e-tests oC10** https://drone.owncloud.com/owncloud/web/27961/11/1
> 
>  To see the trace, please open the link in the console ...

know issue with e2e tests: https://github.com/owncloud/web/issues/7566

**Pull request merged**